sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java

```diff
@@ -182,12 +182,12 @@ private String toKeyVaultSecretName(@NonNull String property) {
             if (property.matches("[a-z0-9A-Z-]+")) {
                 return property.toLowerCase(Locale.US);
             } else if (property.matches("[A-Z0-9_]+")) {
-                return property.toLowerCase(Locale.US).replaceAll("_", "-");
+                return property.toLowerCase(Locale.US).replace("_", "-");
             } else {
                 return property.toLowerCase(Locale.US)
-                    .replaceAll("-", "") // my-project -> myproject
-                    .replaceAll("_", "") // my_project -> myproject
-                    .replaceAll("\\.", "-"); // acme.myproject -> acme-myproject
+                    .replace("-", "") // my-project -> myproject
+                    .replace("_", "") // my_project -> myproject
+                    .replace(".", "-"); // acme.myproject -> acme-myproject
             }
         } else {
```

Comment:
please add more test cases for this

Reply:
After closely looking the test cases, I think the legacy tests case can cover this.  Actually, before this fix mentioned, the test fails. For lucky, we passed it after fix.